xio-core/src/test/java/com/xjeffrose/xio/http/ClientConnectionManagerIntegrationTest.java

```diff
@@ -0,0 +1,95 @@
+package com.xjeffrose.xio.http;
+
+import static com.xjeffrose.xio.helpers.TlsHelper.getKeyManagers;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.xjeffrose.xio.SSL.TlsConfig;
+import com.xjeffrose.xio.bootstrap.ClientChannelConfiguration;
+import com.xjeffrose.xio.client.ClientConfig;
+import com.xjeffrose.xio.helpers.TlsHelper;
+import com.xjeffrose.xio.test.OkHttpUnsafe;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.util.concurrent.Future;
+import java.io.File;
+import java.util.Arrays;
+import okhttp3.Protocol;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ClientConnectionManagerIntegrationTest extends Assert {
+
+  private class HollowChannelHandler extends SimpleChannelInboundHandler<String> {
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, String msg) throws Exception {}
+  }
+
+  private ClientConnectionManager subject;
+  private MockWebServer server;
+
+  private ClientConnectionManager subjectFactory(boolean shouldSucceed) {
+    ClientChannelConfiguration channelConfiguration =
+        new ClientChannelConfiguration(new NioEventLoopGroup(), NioSocketChannel.class);
+
+    File configFile;
+    if (shouldSucceed) {
+      configFile = new File("src/test/resources/ClientConnectionManagerTestSucceed.conf");
+    } else {
+      configFile = new File("src/test/resources/ClientConnectionManagerTestFail.conf");
+    }
+    Config config = ConfigFactory.parseFile(configFile);
+    ClientConfig clientConfig = new ClientConfig(config);
+    ClientState clientState = new ClientState(channelConfiguration, clientConfig);
+
+    ClientChannelInitializer clientChannelInit =
+        new ClientChannelInitializer(clientState, () -> new HollowChannelHandler(), null);
+    return new ClientConnectionManager(clientState, clientChannelInit);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (server != null) {
+      server.close();
+    }
+  }
+
+  @Test
+  public void testInitialConditions() {
+    subject = subjectFactory(true);
+    assertEquals(subject.connectionState(), ClientConnectionState.NOT_CONNECTED);
+  }
+
+  @Test
+  public void testConnectingFailingConnection() throws Exception {
+    subject = subjectFactory(false);
+    // don't set up fake origin backend server so we can connect to it
+    Future<Void> connectionResult = subject.connect();
+    assertEquals(ClientConnectionState.CONNECTING, subject.connectionState());
+    connectionResult.awaitUninterruptibly(60000);
```

Comment:
60 seconds is in the client configuration for conn timeout ... maybe lower it in the test config

Reply:
yeah so, it takes up to that long for the connection to time out, we could configure the client pipeline to have a configurable timeout, but that should be done in another ticket since that is non-zero amount of work.

Reply:
I will try an invalidly formed ip address to see if i can get the channel connect to fail faster

Reply:
ok using a bogus ip address lets us fail the connection faster, will update pr

Reply:
updated